File: Interview_Questions/reverse_array.py
# Reverse an array without using any additional data structures

def reverse_array(arr):
    start = 0
    end = len(arr)-1

    while start < end:
        arr[start], arr[end] = arr[end], arr[start]
        start += 1
        end -= 1

arr = [10, 20, 30, 40, 50, 60]
print("\n")
print("Original Array:", arr)
reverse_array(arr)
print("Reversed Array:", arr)

File: Interview_Questions/reverse_array.c
// Reverse an array without using any additional data structures

#include <stdio.h>

void reverseArray(int arr[], int size) {
    int start = 0;
    int end = size -1;

    while (start < end) {
        int temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;

        start++;
        end--;
    }
}

int main() {
    int arr[] = {10, 20, 30, 40, 50, 60};
    int size = sizeof(arr) / sizeof(arr[0]);

    printf("Original Array:");
    for(int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    reverseArray(arr, size);
    printf("\nReversed Array:");
    for(int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
    return 0;
}